runtime/src/bank/check_transactions.rs

```diff
@@ -221,11 +246,32 @@ mod tests {
             Some(&custodian_pubkey),
             &nonce_hash,
         ));
-        let nonce_account = bank.get_account(&nonce_pubkey).unwrap();
+
+        // set a spurious lamports_per_signature value
+        let mut nonce_account = bank.get_account(&nonce_pubkey).unwrap();
         let nonce_data = get_nonce_data_from_account(&nonce_account).unwrap();
+        nonce_account
+            .set_state(&NonceVersions::new(NonceState::new_initialized(
+                &nonce_data.authority,
+                nonce_data.durable_nonce,
+                stale_lamports_per_signature,
+            )))
+            .unwrap();
+        bank.store_account(&nonce_pubkey, &nonce_account);
+
+        let nonce_account = bank.get_account(&nonce_pubkey).unwrap();
+        let current_lamports_per_signature = bank.get_lamports_per_signature();
```

Comment:
So this test should be using the lps value from `last_blockhash_and_lamports_per_signature`